homeassistant/components/n26/sensor.py

```diff
@@ -30,15 +30,20 @@
 def setup_platform(
         hass, config, add_entities, discovery_info=None):
     """Set up the N26 sensor platform."""
-    api_data = hass.data[DOMAIN][DATA]
+    api_list = hass.data[DOMAIN][DATA]
```

Comment:
Please add a guard clause here that checks if discovery_info is None and return if so.

Reply:
I didn't understand the meaning of this check - we're probably never use discovery_info later. Can you explain shortly?

Reply:
When loading the platform via discovery from the component we pass an empty dict as discovery_info. When the platform is loaded from configuration.yaml, if the user would have a misconfiguration, the discovery_info would be `None`. So we check for `None` to make sure we only go ahead with platform setup if it's loaded via discovery from the component.

Reply:
Done